corehq/util/es/interface.py

```diff
@@ -113,7 +115,28 @@ def scroll(self, scroll_id=None, body=None, params=None, **kwargs):
         return results
 
     def scan(self, index_alias, query, doc_type):
```

Comment:
I think it might be better to just stop using scan entirely and switch to scroll now.  It looks like the only place this is used is in the tragically named `scroll_query`:
https://github.com/dimagi/commcare-hq/blob/e69bf0e66ff3621662f8bea85a9cec7ba3e3ba50/corehq/elastic.py#L235-L238
This in turn is only used in `ESQuery.scroll`:
https://github.com/dimagi/commcare-hq/blob/e69bf0e66ff3621662f8bea85a9cec7ba3e3ba50/corehq/apps/es/es_query.py#L235-L239

This definitely expects to support sorting.  In fact, I just checked and saw at least two places that call that with a sort param - notably exports.  Does this mean that that param has just been ignored thus far?  Could that `ESQuery.scroll` method instead be switched to use an actual scroll query?  If a sort param isn't provided, we should be able to default to `_doc`, which would preserve the behavior for unordered queries.

This would make the query changes happen now, and not as part of a library upgrade, though I think that's actually an advantage.

Reply:
Thanks @esoergel. Looking into that `scroll_query` usage was at the top of my todo list for today.  I've been suspicious of the fact that `ESQuery.scroll` (and thus, most/all ES models) is based on this `scan()` helper. Certainly needs further investigation.

> Does this mean that that param has just been ignored thus far?

This is a possibility, but I have not confirmed it.  I have not tested what happens in a _sorted_ search query where the `search_type='scan'` parameter is added (this is the current HQ implementation of `ESQuery.scroll`).  Using `search_type='scan'` seems to imply an unordered result, so I don't know what the ES engine does when you specify it on a query with `sort`.

Upon switching to the `elasticsearch5` library (without this PR), any use of (the current implementation of) `ESQuery.scroll` to perform a sorted query will resulted in the `sort` field being silently clobbered to `_doc` due to our `scroll_query`'s use of the `scan()` helper.

> Could that ESQuery.scroll method instead be switched to use an actual scroll query?

Yes, and it should be. I believe it has always been true that using `sort` in a query with the `scan()` function negates the benefits of that helper (because specifying `search_type='scan'` implies an unordered result).